build_shared.sh

```diff
@@ -0,0 +1,78 @@
+set -e
+# Check if mkdtimg tool exist
+if [ ! -f "$MKDTIMG" ]; then
+    echo "mkdtimg: File not found!"
+    echo "Building mkdtimg"
+    export ALLOW_MISSING_DEPENDENCIES=true
+    make mkdtimg
+fi
+
+cd "$KERNEL_TOP"/kernel
+
+echo "================================================="
+echo "Your Environment:"
+echo "ANDROID_ROOT: ${ANDROID_ROOT}"
+echo "KERNEL_TOP  : ${KERNEL_TOP}"
+echo "KERNEL_TMP  : ${KERNEL_TMP}"
+
+for platform in $PLATFORMS; do \
+
+    case $platform in
+        loire)
+            DEVICE=$LOIRE;
+            DTBO="false";;
+        tone)
+            DEVICE=$TONE;
+            DTBO="false";;
+        yoshino)
+            DEVICE=$YOSHINO;
+            DTBO="false";;
+        nile)
+            DEVICE=$NILE;
+            DTBO="false";;
+        ganges)
+            DEVICE=$GANGES;
+            DTBO="false";;
+        tama)
+            DEVICE=$TAMA;
+            DTBO="true";;
+        kumano)
+            DEVICE=$KUMANO;
+            DTBO="true";;
+    esac
+
+    for device in $DEVICE; do \
+        (
+            if [ ! $only_build_for ] || [ $device = $only_build_for ] ; then
+
+                KERNEL_TMP=$KERNEL_TMP-${device}
+                # Keep kernel tmp when building for a specific device or when using keep tmp
+                [ ! "$keep_kernel_tmp" ] && [ ! "$only_build_for" ] &&rm -rf "${KERNEL_TMP}"
+                mkdir -p "${KERNEL_TMP}"
+
+                echo "================================================="
+                echo "Platform -> ${platform} :: Device -> $device"
+                make O="$KERNEL_TMP" ARCH=arm64 CROSS_COMPILE="$CROSS_COMPILE" -j$(nproc) ${BUILD_ARGS} aosp_${platform}_${device}_defconfig
+
+                echo "The build may take up to 10 minutes. Please be patient ..."
+                echo "Building new kernel image ..."
+                echo "Logging to $KERNEL_TMP/build.log"
+                make O="$KERNEL_TMP" ARCH=arm64 CROSS_COMPILE="$CROSS_COMPILE" -j$(nproc) ${BUILD_ARGS} ${CC+"${CC}"} \
+                     >"$KERNEL_TMP"/build.log 2>&1;
+
+                echo "Copying new kernel image ..."
+                cp "$KERNEL_TMP/arch/arm64/boot/Image.gz-dtb" "$KERNEL_TOP/common-kernel/kernel-dtb-$device"
+                if [ $DTBO = "true" ]; then
+                    # shellcheck disable=SC2046
```

Comment:
I know that thats why I disabled the shellcheck warning. I tested script with akari and everything was fine. I can try to add printf to the find command and test.
I find dev environment really an edge case that its ok to support but really its time consuming.

Reply:
That would be nice, just to be sure we aren't running into the (very unlikely) example of a dtbo with spaces.

If you are stressed for time you can always hand the PR over to someone else, while jumping in on the discussion. But to be fair, this one is a massive nitpick :)

Reply:
>If you are stressed for time you can always hand the PR over to someone else, while jumping in on the discussion. But to be fair, this one is a massive nitpick :)

No problem. I just don't like fixing stupid edge cases when they make things ugly like using eval which is something I try to avoid.
It's normal that such rewrites take time.